addons/l10n_ch/tests/test_gen_isr_reference.py

```diff
@@ -0,0 +1,124 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo.tests import common, Form
+
+QR_IBAN = 'CH21 3080 8001 2345 6782 7'
+ISR_SUBS_NUMBER = "01-162-8"
+
+
+class TestGenISRReference(common.SavepointCase):
+    """Check condition of generation of and content of the structured ref"""
+
+    @classmethod
+    def setUpClass(cls):
+        super().setUpClass()
+        cls.env = cls.env(context=dict(cls.env.context, tracking_disable=True))
+        cls.partner = cls.env.ref("base.res_partner_12")
+        cls.bank = cls.env["res.bank"].create(
+            {
+                "name": "Alternative Bank Schweiz AG",
+                "bic": "ALSWCH21XXX",
+            }
+        )
+        cls.bank_acc_isr = cls.env["res.partner.bank"].create(
+            {
+                "acc_number": "ISR",
+                "l10n_ch_isr_subscription_chf": "01-162-8",
+                "bank_id": cls.bank.id,
+                "partner_id": cls.partner.id,
+            }
+        )
+        cls.bank_acc_qriban = cls.env["res.partner.bank"].create(
+            {
+                "acc_number": QR_IBAN,
+                "bank_id": cls.bank.id,
+                "partner_id": cls.partner.id,
+            }
+        )
+
+    def new_form(self):
```

Comment:
Let's avoid duplicating this kind of helper in all our tests (that's what we used to do before, I know ^^). You could probably use this function instead https://github.com/odoo/odoo/blob/13.0/addons/account/tests/account_test_savepoint.py#L361 If you need to add it some optional params, feel free to do it. For simple invoices creation, we want to use this helper as much as possible.

Reply:
Nice didn't know about this one.